Contributing.md

```diff
@@ -0,0 +1,163 @@
+4. **No unnecessary or unrelated changes.** PRs shouldn’t come with random formatting changes, especially in unrelated parts of the code. If there is some refactoring that needs to be done, it should be in a separate PR from a bug fix or feature, if possible.
+5. **Code has appropriate comments.** Code should be commented describing the problem it is solving, not just the technical implementation.
+
+## Developing the CLI
+
+### Quick Start
+
+If you have all of the dependencies setup and are ready to get right into contributing, run the following command after cloneing the repo:
+
+```
+cargo run -- help
+```
+
+This step will install of the dependencies and print out the help command from the CLI. To run develop commands and test them out locally, you can use the `cargo run` command and pass commands and headers like this:
+
+```
+cargo run -- <command name> --<flag>
+```
+
+Cargo will proxy your commands and flags to the built CLI project.
+
+### Deep dive
+
+**Prerequisites:**
+This project is written in Rust and setup as a multi crate (aka package) project to allow us to share common code across multiple packages.  To get started you need to have the Rust toolchain installed. The best way to do this is to visit [the amazing rust site](https://www.rust-lang.org/learn/get-started) and follow their install instructions.
+
+To ensure you are ready, you should be able to run this:
+
+`cargo --version` 
+
+and get something back that looks like this:
+
+`cargo 1.42.0 (86334295e 2020-01-31)`
+
+**Testing the project**
+Rust has great built-in test tooling. You can write tests in-line or under the tests folder for each crate. Tests under the `tests` folder should be integration whereas inline should be unit tests (as I understand from the conventions). You can run `cargo test` to test all of the crates at once.
+
+
+> If you have your editor setup with the Rust extension, you can click the codelens “Test” command to run individual tests right from you editor! How cool is that?!
+
+
+**Opening a Pull Request**
+Once you have fixed a bug, created the next great feature, or improved the project in some other way it is time to open a pull request! To learn more about how to do this, follow [this great guide](https://opensource.guide/how-to-contribute/#opening-a-pull-request). When you open a PR, the pipeline steps will run to ensure the changes will work for users of this library. To learn more about the pipeline, keep reading below!
+
+## Pipelines
+
+This project uses GitHub Actions to run a continuous integration and delivery pipeline. Every code change will be run against a few steps to help keep the project running at its peak ability
+
+* **CLA Check**: If you haven’t signed the Apollo CLA, a bot will comment on your PR asking you to do this
```

Comment:
Should we link to the that here? Small improvement that could definitely be skipped

Reply:
I still need to turn it on haha